eval/document.go

```diff
@@ -74,6 +75,21 @@ func (doc *Document) RenderContent(ctx context.Context, docDataCtx plugindata.Ma
 		return nil, nil, diags
 	}
 
+	// verify required vars
+	if len(doc.RequiredVars) > 0 {
+		vars, varsPresent := docDataCtx["vars"].(plugindata.Map)
+		for _, reqVar := range doc.RequiredVars {
+			if !varsPresent || vars[reqVar] == nil {
+				diags.Append(&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Missing required variable",
+					Detail:   "document." + name + " block requires '" + reqVar + "' var which is not set.",
+					Subject:  nil,
```

Comment:
And here you would be able to use `Subject: doc.Source.Raw.Block.Range()` to point the user at the offending block